### ISSUE TITLE
Apache HTTP Client connection metrics

### ISSUE THREAD

**Issue by j0hnc0yne** [micrometer-metrics/micrometer#4956]
**Please describe the feature request.**

The Apache HTTP Client metrics do not include the timings and count for when new connections are created.  Reactor Netty [has these](https://projectreactor.io/docs/netty/1.1.16-SNAPSHOT/reference/index.html#_metrics_6) as `reactor.netty.http.client.connect.time` and `reactor.netty.http.client.tls.handshake.time`.  

**Rationale**

These metrics are useful to see both how often new connections are created as well as how long those take.  Using that information can provide better insight into configuring the connection pool.

**Comment by shakuzen** [comment]
Thanks for opening the request. I can see how this would be useful. We currently get all of the connection metrics from `ConnPoolControl`. Wrapping PoolingHttpClientConnectionManager sounds a bit more involved and potentially more maintenance challenges. It would be better if Apache HTTP Client exposed some API that made it easy for us to instrument connection creation. What do you think?

**Comment by j0hnc0yne** [comment]
> Thanks for opening the request. I can see how this would be useful. We currently get all of the connection metrics from `ConnPoolControl`. Wrapping PoolingHttpClientConnectionManager sounds a bit more involved and potentially more maintenance challenges. It would be better if Apache HTTP Client exposed some API that made it easy for us to instrument connection creation. What do you think?

Thanks for the reply.  That does make sense. So are you suggesting to submit a Feature Request to the Apache HttpComponents project?

**Comment by shakuzen** [comment]
Yes, if there were some API available for a listener/callback to instrument connection creation, that would enable this feature to be implemented.

**Comment by shakuzen** [comment]
See https://github.com/micrometer-metrics/micrometer/issues/7158. There is now instrumentation in a dedicated module in the Apache HttpClient project for Micrometer instrumentation. According to [the documentation](https://hc.apache.org/httpcomponents-client-5.6.x/observation.html) there are some metrics for connections but perhaps not exactly what was requested here. It would be best to request any enhancements for this to the Apache HttpClient project. For that reason, I'll close this issue. Feel free to ping me if I can help with anything Micrometer-related in doing so.
